library/src/scala/tasty/Reflection.scala

```diff
@@ -1336,20 +1336,20 @@ trait Reflection extends reflect.Types { reflectSelf: CompilerInterface =>
   //   TYPES   //
   ///////////////
 
-  /** Returns the type (Type) of T */
-  def typeOf[T](using qtype: scala.quoted.Type[T], ctx: Context): Type =
-    qtype.asInstanceOf[scala.internal.quoted.Type[T]].typeTree.asInstanceOf[TypeTree].tpe
-
-
   // ----- Types ----------------------------------------------------
 
   given (using ctx: Context) as TypeTest[Type, Type] = reflectSelf.Type_TypeTest
   given TypeOps as Type.type = Type
 
   object Type:
 
-    def apply(clazz: Class[_])(using ctx: Context): Type =
-      reflectSelf.Type_apply(clazz)
+    /** Returns the type or kind (Type) of T */
+    def of[T <: AnyKind](using qtype: scala.quoted.Type[T], ctx: Context): Type =
+      qtype.asInstanceOf[scala.internal.quoted.Type[TypeTree]].typeTree.tpe
```

Comment:
```suggestion
      qtype.unseal
```

Reply:
Here we cannot do this as we do not know which is the quote context that owns this reflect instance. There is another PR were I make this change possible.